rstream/producer.py

```diff
@@ -190,21 +226,54 @@ async def publish_batch(
 
         return publishing_ids
 
-    async def publish(
+    async def send_wait(
         self,
         stream: str,
         message: MessageT,
-        sync: bool = True,
         publisher_name: Optional[str] = None,
     ) -> int:
-        publishing_ids = await self.publish_batch(
+        publishing_ids = await self._send_batch(
             stream,
             [message],
-            sync=sync,
+            sync=True,
             publisher_name=publisher_name,
         )
         return publishing_ids[0]
 
+    async def send(
+        self,
+        stream: str,
+        message: MessageT,
+        publisher_name: Optional[str] = None,
+    ):
+
+        if stream not in self._buffered_messages:
+            self._buffered_messages[stream] = []
+        async with self._lock:
+            await self._get_or_create_publisher(
+                stream, send_batch_enabled=True, publisher_name=publisher_name
+            )
+        async with self._buffered_messages_lock:
+            self._buffered_messages[stream].append(message)
+
+        await asyncio.sleep(0)
```

Comment:
Hmm do we really need it here? This statement only makes sense if there's no other `__await__` statements in a function and we want to force a context switch anyway

Reply:
@qweeze. I would like to discuss about this point because I think it's the most important one. I agree with you on this but apparently on the tests made it appears that if I don't force the context switch there, all the cpu get taken by the calling process which is continuously calling send() and the background thread don't get never activated. I'm not really an asyncio expert but I think it is related on what explained on this article here: https://towardsdatascience.com/asyncio-is-not-parallelism-70bfed470489.
Also I see around that using asyncio.sleep(0) is not considered a bad practice with this library after all (https://superfastpython.com/asyncio-sleep/).

Reply:
I was wondering if starting the task before we start looping to send() messages may improve the situation without forcing us to use the sleep. Will do few tests.

Reply:
No I'm having the same behavior starting the thread even before looping for send(), it seems like the sleep() is necessary.

Reply:
Yes, that's the expected behavior because asyncio's concurrency model is different from threading (cooperative vs preemptive multitasking) - the event loop can only switch to another coroutine when the current coroutine is suspended waiting for future result. That's why any long-running sync code effectively blocks the event loop and considered a bad thing when writing asyncio apps

In general as a library authors we can't prevent users from blocking the event loop - for example, someone can just put `time.sleep(5)` in their code and it will block our background `_timer` task. But it's expected that a programmer is aware that they should avoid blocking code.

In this specific case with `send` method I think the problem is that the method is `async`, but it actually awaits IO only on the first call, when `self._publishers` and `self._clients` are not initialized. And on subsequent calls all the operations are sync, which makes the call effectively blocking. So I agree that we can use `asyncio.sleep(0)` here for consistency.

Or alternatively we can use `asyncio.Queue` instead of a list for buffering messages, that way we'll have "natural" context switch, and also a way to limit the number of messages in a buffer. (also `_buffered_messages_lock` will become unnecessary)

Reply:
Thanks for the feedback.
Ok great we will try with `asyncio.Queue`. In case of problems with `asyncio.Queue` we will merge the current version if there isn't any other feedback.

Reply:
I made a few tests trying to use asyncio.Queue but it seems like performance deteriorates even when removing the old lock we used for the list. Also the send uses internally the send_batch which takes in input a normal list[]. So some sort of conversion is needed. I think for the moment we can keep it with the sleep(0) and open a new issue to better investigate the usage of asyncio.Queue